action_server/frontend/src/views/actions/components/ActionRun.tsx

```diff
@@ -64,6 +77,64 @@ export const ActionRun: FC<Props> = ({ action, actionPackage }) => {
     [action, actionPackage, formData, apiKey, serverConfig],
   );
 
+  const onCodeChange = useCallback(
+    (code: string) => {
+      try {
+        setErrorJSON(undefined);
+        const rawInputData = JSON.parse(code as string);
+        const tempForm = [...formData];
+
+        Object.entries(rawInputData).forEach(([key, val]) => {
+          const foundFormItem = tempForm.find(
+            (elem) => elem.name.toLowerCase() === key.toLowerCase(),
+          );
+          if (foundFormItem) {
+            const foundFormItemIndex = tempForm.findIndex(
+              (elem) => elem.name.toLowerCase() === key.toLowerCase(),
+            );
+            tempForm[foundFormItemIndex] = { ...foundFormItem, value: val as string };
+          }
+        });
+
+        debounce(() => setFormData([...tempForm]), 750)();
+      } catch (e) {
+        setErrorJSON('Validating JSON syntax failed. Please verify the input and try again.');
+      }
+    },
+    [formData],
+  );
+
+  const rawJSONInput = useMemo(() => {
+    try {
+      if (formData.length === 0) {
+        return 'No input data...';
+      }
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const output: { [key: string]: any } = {};
+      formData.forEach((prop) => {
+        switch (prop.property.type) {
+          case 'boolean':
+            output[prop.name.toLowerCase()] = prop.value.toLowerCase() === 'true';
+            break;
+          case 'number':
+          case 'integer':
+            output[prop.name.toLowerCase()] = parseFloat(prop.value);
+            break;
+          case 'object':
+            output[prop.name.toLowerCase()] = JSON.parse(prop.value);
+            break;
+          case 'enum':
+          default:
+            output[prop.name.toLowerCase()] = prop.value;
+            break;
+        }
+      });
+      return JSON.stringify(output, null, 4);
+    } catch (e) {
+      return 'There was an error while parsing form data...';
+    }
+  }, [formData]);
```

Comment:
The JSON should be created in a nested manner – `FormData` is in a flattened form.
There is an helper function that can be used for this at https://github.com/robocorp/robocorp/blob/f98345ef852a863a2fbb36b279086c2138700d41/action_server/frontend/src/lib/formData.ts#L133

Reply:
Done.